webapp/src/js/app.js

```diff
@@ -134,21 +134,22 @@ _.templateSettings = {
     contacts: 'can_view_contacts',
     analytics: 'can_view_analytics',
     reports: 'can_view_reports',
+    'reports.edit': 'can_update_reports'
   };
 
   var getRequiredPermission = function(route) {
-    var baseRoute = route.split('.')[0];
+    var baseRoute = route in ROUTE_PERMISSIONS ? route : route.split('.')[0];
     return ROUTE_PERMISSIONS[baseRoute];
   };
 
   // Detects reloads or route updates (#/something)
   angular.module('inboxApp').run(function($state, $transitions, Auth) {
-    $transitions.onStart({}, function(trans) {
+    $transitions.onBefore({}, function(trans) {
       if (trans.to().name.indexOf('error') === -1) {
         var permission = getRequiredPermission(trans.to().name);
         if (permission) {
-          Auth(permission).catch(function() {
-            $state.go('error', { code: 403 });
+          return Auth(permission).catch(function() {
+            return $state.target('error', { code: 403 });
```

Comment:
This is the documented way to redirecting transitions (https://ui-router.github.io/guide/transitionhooks#redirecting-a-transition). The way it was before, the blocked transition was not stopped, but overwritted by the new transition to `error`, resulting in this error message in the console:
![image](https://user-images.githubusercontent.com/35681649/55059216-1418a300-5077-11e9-9b34-54591ef5ff89.png)

Reply:
Nice catch.